core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.ts

```diff
@@ -1142,22 +1127,18 @@ export class ConversationSkinComponent {
         this.nextCard.getStateName());
     }
 
-    if (this.CHECKPOINTS_FEATURE_IS_ENABLED) {
-      // We do not store checkpoints progress for iframes hence we do not
-      // need to consider redirecting the user to the most recently
-      // reached checkpoint on exploration initial load in that case.
-      if (!this.isIframed && !this._editorPreviewMode &&
-          !this.explorationPlayerStateService.isInQuestionPlayerMode()) {
-        // Navigate the learner to the most recently reached checkpoint state.
-        this._navigateToMostRecentlyReachedCheckpoint();
-      }
+    // We do not store checkpoints progress for iframes hence we do not
+    // need to consider redirecting the user to the most recently
+    // reached checkpoint on exploration initial load in that case.
+    if (!this.isIframed && !this._editorPreviewMode &&
+        !this.explorationPlayerStateService.isInQuestionPlayerMode()) {
+      // Navigate the learner to the most recently reached checkpoint state.
+      this._navigateToMostRecentlyReachedCheckpoint();
     }
 
     this.focusManagerService.setFocusIfOnDesktop(focusLabel);
     this.loaderService.hideLoadingScreen();
-    if (!this.CHECKPOINTS_FEATURE_IS_ENABLED) {
-      this.hasFullyLoaded = true;
-    }
+    this.hasFullyLoaded = true;
```

Comment:
Not sure you did this correctly -- there's a ! before the condition. This might be causing your failing test issues.

Reply:
I think this check explains the [issue](https://github.com/oppia/oppia/issues/18341) that I opened yesterday.
Basically if we do not do `this.hasFullyLoaded = true;` then the preview tab is not visible because the first line in the HTML is as follows:
```
<div role="main" class="oppia-conversation-skin-animate-cards-container position-relative" *ngIf="hasFullyLoaded">
```
Also, I think this should be present here because note that we have `this.loaderService.hideLoadingScreen();` just few lines below, so I think when we hide the loading screen we already have the data that we require?

Reply:
Sorry, I have no idea (and don't have time to debug it now, apologies). One thing you might want to try is looking at the PR before checkpoints was introduced and seeing what happened there -- that might give you a way to double-check.

Reply:
@seanlip,
Thanks for the suggestion @seanlip, I think previously the changes were similar to what I have made here and then in this PR [here](https://github.com/oppia/oppia/pull/18110/files) the `!this.CHECKPOINTS_FEATURE_IS_ENABLED` check was added which was merged around a month back.
I think this change would be fine. Let me know what you think.
Thanks

Reply:
Yup I think I agree with you.